Document disable default providers (#1050)
pulumi/pulumi-hugo: themes/default/content/docs/intro/concepts/resources/providers.md

{{% /choosable %}}

{{< /chooser >}}

> While default providers are enabled by default, they can be disabled on a per stack basis. You might want to do this if passing an Explicit Providers is required for all created resources. For example, to disable the `aws` provider, you can run

Reviewer: ```suggestion
While default providers are enabled by default, they [can be disabled]({{< relref "/docs/intro/concepts/config#special-configuration-options" >}}) on a per stack basis. You might want to do this if passing an explicit provider is required for all created resources. For example, to disable the `aws` provider, you could run:
```

```sh
$ pulumi config set --path 'pulumi:disable-default-providers[0]' aws
```

> If you want to also disable the `kubernetes` default provider, run

Reviewer: ```suggestion
If you want to also disable the `kubernetes` default provider, run:
```

> If you want to also disable the `kubernetes` default provider, run

Reviewer: ```suggestion
If you wanted to also disable the `kubernetes` default provider as well, in addition to the `aws` default provider, you could run:
```

```sh
$ pulumi config set --path 'pulumi:disable-default-providers[1]' kubernetes
```

To disable all default providers, use `*` as the package name:

```sh

> $ pulumi config set --path 'pulumi:disable-default-providers[2]' '*'

Reviewer: Would this maybe make more sense (assuming you'd most likely define `*` as the sole entry)?

```suggestion
$ pulumi config set --path 'pulumi:disable-default-providers[0]' '*'
```

{{% /choosable %}}

{{< /chooser >}}

> While default providers are enabled by default, they [can be disabled]({{< relref

Reviewer 2: Suggest we put this under a section header like `## Disabling Default Providers`.

"/docs/intro/concepts/config#special-configuration-options" >}}) on a per stack basis. You might want to do this if

> passing an explicit providers is required for all created resources. For example, to disable the `aws` provider, you can

Reviewer: ```suggestion
passing an explicit provider is required for all created resources. For example, to disable the `aws` provider, you can
```

> passing an explicit providers is required for all created resources. For example, to disable the `aws` provider, you can

Reviewer 2: Perhaps worth expanding slightly on the reasons this may be desirable? "To avoid accidentally depending on the default provider", "to ensure your programs must be explicit about which provider they will use".

> run

Reviewer 2: ```suggestion
run:
```
